test/components/labelTests.ts

```diff
@@ -1,198 +1,217 @@
-    let svg = TestMethods.generateSVG(400, 400);
-    let label = new Plottable.Components.TitleLabel("foo");
-    label.renderTo(svg);
-    label.text("");
-    assert.strictEqual(label.width(), 0, "width updated to 0");
-    svg.remove();
+  describe("Basic Usage", () => {
+
+    it("only accepts strings as input for text", () => {
+      const label = new Plottable.Components.Label();
+      // HACKHACK #2661: Cannot assert errors being thrown with description
+      (<any> assert).throws(() => label.text(<any> 23), Error,
+        "Label.text() only takes strings as input", "error on inputing number");
+      (<any> assert).throws(() => label.text(<any> new Date()), Error,
+        "Label.text() only takes strings as input", "error on inputing Date");
+      assert.doesNotThrow(() => label.text("string"), Error, "text() accepts strings");
+      assert.strictEqual(label.text(null), "string", "text(null) returns text string");
+    });
+
+    it("converts angle to stay within -180 and 180", () => {
+      const label360 = new Plottable.Components.Label("noScope", 360);
+      assert.strictEqual(label360.angle(), 0, "angles are converted to range [-180, 180] (360 -> 0)");
+      const label270 = new Plottable.Components.Label("turnRight", 270);
+      assert.strictEqual(label270.angle(), -90, "angles are converted to range [-180, 180] (270 -> -90)");
+      const labelNeg270 = new Plottable.Components.Label("turnRight", -270);
+      assert.strictEqual(labelNeg270.angle(), 90, "angles are converted to range [-180, 180] (-270 -> 90)");
+    });
+
+    it("throws error on invalid angles", () => {
+      let badAngle = 10;
+      (<any> assert).throws(() => new Plottable.Components.Label("foo").angle(badAngle), Error,
+        `${badAngle} is not a valid angle for Label`, "only accept -90/0/90 for angle");
+      (<any> assert).throws(() => new Plottable.Components.Label("foo", badAngle), Error,
+        `${badAngle} is not a valid angle for Label`, "only accept -90/0/90 for angle");
+    });
+
+    it("positions centered text in a table properly", () => {
+      const svgWidth = 400;
+      const svg = TestMethods.generateSVG(svgWidth);
+      const label = new Plottable.Components.Label("X");
+      const t = new Plottable.Components.Table().add(label, 0, 0)
+                                        .add(new Plottable.Component(), 1, 0);
+      t.renderTo(svg);
+      const textTranslate = d3.transform(label.content().select("g").attr("transform")).translate;
+      const eleTranslate = d3.transform((<any> label)._element.attr("transform")).translate;
+      const textWidth = Plottable.Utils.DOM.elementBBox(label.content().select("text")).width;
+      assert.closeTo(eleTranslate[0] + textTranslate[0] + textWidth / 2, svgWidth / 2,
+        window.Pixel_CloseTo_Requirement, "label is centered");
+      svg.remove();
+    });
+
+    it("pads reacts under align", () => {
```

Comment:
It's not clear what this test is checking for -- can we use "it <verb>" wording and give it a more descriptive name?